test/run-tests.c

```diff
@@ -177,5 +179,14 @@ static int maybe_run_test(int argc, char **argv) {
     return spawn_stdin_stdout();
   }
 
+#ifndef _WIN32
+  if (strcmp(argv[1], "spawn_helper10") == 0) {
+    int fd = open("/dev/sda", O_RDONLY);
+    ASSERT(-1 != fd);
+    close(fd);
+    return 1;
+  }
+#endif  /* !_WIN32 */
```

Comment:
I don't think this is portable. Not all the unixes have user `disk` defined nor `/dev/sda`. Maybe a solution would be passing the additional group_id as parameter and check the child process' secondary group is the same.